cli/explain/explain.go

```diff
@@ -138,13 +173,87 @@ func diff(oldPath, newPath string) (*spawn_diff.DiffResult, error) {
 	return compactgraph.Diff(oldGraph, newGraph)
 }
 
-func readGraph(path string) (*compactgraph.CompactGraph, error) {
-	f, err := os.Open(path)
+var uuidPattern = regexp.MustCompile("^(?:.*/invocation/)?([0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12})$")
+
+func openLog(pathOrId string) (io.ReadCloser, error) {
+	f, err := os.Open(pathOrId)
+	if err == nil {
+		return f, nil
+	} else if !os.IsNotExist(err) || !uuidPattern.MatchString(pathOrId) {
+		return nil, err
+	}
+	matches := uuidPattern.FindStringSubmatch(pathOrId)
+	invocationId := matches[1]
+	// This is an invocation ID, try to fetch its corresponding log.
+	ctx, err := login.WithApiKeyInteractive(context.Background(), "", false)
 	if err != nil {
 		return nil, err
 	}
-	defer f.Close()
-	return compactgraph.ReadCompactLog(f)
+	backend, err := storage.GetLastBackend()
```

Comment:
What should be the behavior if this is the first request from the CLI? i.e. if last backend is not set. I think it's reasonable to the use the prod URL

Reply:
If we use a default URL as a fallback, the API key will be sent to that URL. In case of a security conscious self-hosted user, that could be considered problematic even if the default is our prod URL. But it's clear that from the point of view of usability, not failing in that case would be better. What do you think?

Reply:
The CLI already uses our prod URL as the default in most places, so I think it's okay to continue doing that. I think it's fair to expect a security conscious user to rotate a "leaked" API key if they realize they've sent it to the wrong server

Reply:
Sounds reasonable. I extracted the default backend into a constant for `//cli`.

Reply:
Yes, I will add that in https://github.com/buildbuddy-io/buildbuddy/pull/8217 (I could also merge it into this PR if you prefer).